src/fabric.c

```diff
@@ -591,6 +600,25 @@ static void ofi_ini_dir(const char *dir)
 }
 #endif
 
+static char *ofi_handle_prov_dl(const char *orig_prov_name)
+{
+	char *prov_name = strdup(orig_prov_name), *iter;
+
+	if (!prov_name)
+		return NULL;
+
+	iter = prov_name;
+
+	for (; *iter; ++iter)
+		*iter = tolower(*iter);
+
+	return prov_name;
+}
+
+#define FI_PROV_CHECK_DL(prov_name)		\
+	((HAVE_ ## prov_name ## _DL) ?		\
+	 ofi_handle_prov_dl(#prov_name) : NULL)
```

Comment:
We don't need the above changes

Reply:
this is used to handle only such providers for which HAVE_<prov>_DL was defined, i.e. we can skip static providers
what you will suggest to keep this filtering with ordered_prov_names list ?

Reply:
You don't want to skip static providers, because the DL build of that provider is how it can be updated in the field without replacing all of libfabric.

Consider that libfabric is built with verbs, tcp, usnic, efa, psm2, etc. as static.  That version ships with the distro.  A critical bug is then found in the psm2 provider.  The DL version allows replacing only the psm2 provider without changing what the distro is shipping for other providers.

Similarly, there may be a DL provider that is shipped by a third party which the libfabric build was not aware of.  This allows a third party to add support for their provider against a distro shipped libfabric.

We need to continue to support both of these models.

I'm okay adding the additional checks against the standard library path in order to pick-up updated providers that we know about.  But that can be against all known providers, and is not limited by how libfabric was built.

Btw, the main reason why we allow building a provider as a DL is so that we have a way to generate the DL provider, and can test that it works.  It's not intended as a way to identify which providers as DL only.

Reply:
I see. Maybe I described special case when OFI is built w/o any static provider, all providers are dynamic ones intentionally -  it looks like custom case which we need to skip from this PR.